apidef/oas/schema/x-tyk-gateway.md

```diff
@@ -965,6 +968,65 @@ Path file path for the template.
 Body base64 encoded representation of the template.
 
 
+### **URLRewrite**
+
+**Field: `enabled` (`boolean`)**
+Enabled enables URL rewriting if set to true.
+
+**Field: `pattern` (`string`)**
+Pattern is the regular expression against which the request URL is compared for the primary rewrite check.
+If this matches the defined pattern, the primary URL rewrite is triggered.
+
+**Field: `rewriteTo` (`string`)**
+RewriteTo specifies the URL to which the request shall be rewritten if the primary URL rewrite is triggered.
+
+**Field: `triggers` (`[]`[URLRewriteTrigger](#urlrewritetrigger))**
+Triggers contain advanced additional triggers for the URL rewrite.
+The triggers are processed only if the requested URL matches the pattern above.
+
+
+### **URLRewriteTrigger**
+
+**Field: `condition` ([](#))**
+Condition indicates the logical combination that will be applied to the rules for an advanced trigger:
+
+- Value `any` means any of the defined trigger rules may match
+- Value `all` means all the defined trigger rules must match
+
+**Field: `rules` (`[]`[URLRewriteRule](#urlrewriterule))**
+Rules contain individual checks that are combined according to the `condition` to determine whether the URL rewrite will be triggered.
+If empty, the trigger is ignored.
+
+**Field: `rewriteTo` (`string`)**
+RewriteTo specifies the URL to which the request shall be rewritten if indicated by the combination of `condition` and `rules`.
+
+
+### **URLRewriteRule**
+
+**Field: `in` ([](#))**
+In specifies one of the valid inputs for URL rewriting.
+By default, it uses `url` as the input source.
+The following values are valid:
+
+- `url`, match pattern against URL
+- `query`, match pattern against named query parameter value
+- `path`, match pattern against named path parameter value
+- `header`, match pattern against named header value
+- `sessionMetadata`, match pattern against session metadata
+- `requestBody`, match pattern against request body
+- `requestContext`, match pattern against request context
+
+**Field: `name` (`string`)**
+Name is the index in the input identified in `in` that should be used to locate the value for this rule. `Name` is ignored for `InputRequestBody`rules as it contains only a single value, while the others are objects.
```

Comment:
If I've understood correctly, this is actually only used for the following values in the `in` field: `query`, `path`, `header` (i.e. where in the docs above we've said "match pattern against named ... parameter value").

So perhaps (and Github isn't giving me the option to "suggest"):

Name is the index in the input identified in `in` that should be used to locate the value for this rule. `Name` is only relevant for `query`, `path` and `header` values in the `in` field, for all other values it is ignored.